### PR TITLE
297 baisc dynamic example

### PR DESCRIPTION
**Pull request by kaison428**
https://github.com/madil4/awatif/assets/38864087/89110646-1406-4d17-9762-ddc693486e2f

Implemented a basic dynamic analysis example with 2-bar pendulum

### CODE REVIEW

File: awatif-data-structure/src/analysis.ts [hunk 1]
```diff
@@ -34,6 +35,13 @@ export type LoadAssignment = {
     | [number, number, number, number, number, number];
 };
 
+export type LumpedMassAssignment = {
```
Comment (madil4) [inline awatif-data-structure/src/analysis.ts:38]: I prefer to assign the `density` to the element instead

Comment (madil4) [inline awatif-data-structure/src/analysis.ts:38]: I prefer to assign the density to the element instead

File: awatif-ui/src/App.ts [hunk 2]
```diff
@@ -33,11 +36,16 @@ export function app({ model, parameters, onParameterChange, settings }: App) {
...
+    dynamic: van.state(settings?.dynamic ?? false),
+    dynamicSettings: van.state(
+      settings?.dynamicSettings ?? { time: 1, timeStep: 1 }
+    ),
   };
 
   // update
   Viewer(modelState, settingsState);
   Settings(settingsState);
+  if (settings?.dynamic ?? false) Timeline(modelState, settingsState);
```
Comment (madil4) [inline awatif-ui/src/App.ts:48]: We can check if there is a dynamic result by inspecting the `analysisResult` state, like so:
```modelState.val.analysisResults.position.size > 1 ```
This way the app is smart enough to show the slider or not

File: awatif-ui/src/Viewer.ts [hunk 3]
```diff
@@ -16,13 +16,17 @@ import { ElementResults } from "./objects/ElementResults";
 import { NodeResults } from "./objects/NodeResults";
 
 export function Viewer(model: ModelState, settings: SettingsState) {
+  // config
+  let playerHeight = 0;
```
Comment (madil4) [inline awatif-ui/src/Viewer.ts:20]: There is no need to change the height of the viewer if you used the position attribute in CSS like so:
```
.player {
  position: absolute
  bottom: 0px
}
```
It is cleaner to handle the positioning in the CSS file where it belong

File: examples/basic-dynamic/analyzeDynamically.ts [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+  PositionResult,
+  AnalysisResults,
+  LumpedMassAssignment,
+  PropertyAssignment,
+} from "../../awatif-data-structure";
+import { Node, Element, Assignment, Parameters } from "../../awatif-ui/";
+import { getTransformationMatrix } from "../../awatif-ui/src/utils/getTransformationMatrix.ts";
+import * as math from "mathjs";
+
+export function analyzeDynamically(
```
Comment (madil4) [inline examples/basic-dynamic/analyzeDynamically.ts:11]: Did you validate the correctness of the algorithm with the Python code you showed me earlier? the animation should match the double-pendulum result

**Review by madil4 — CHANGES_REQUESTED** [review]
I liked the quality of the code and the result looks super cool, good job!

I have a few comments before merging into the main branch. After that, we can expose some parameters to the user such as (density, stiffness, timestep...etc) and then we can release the public.

File: examples/basic-dynamic/analyzeDynamically.ts (hunk 4 above)
Comment (kaison428) [reply examples/basic-dynamic/analyzeDynamically.ts:11]: https://github.com/madil4/awatif/assets/38864087/735df4a0-5e0b-4acf-b90b-4772bcc75d62

It does based on multiple comparisons. We can do a more thorough validation if needed.

File: awatif-ui/src/Viewer.ts (hunk 3 above)
Comment (kaison428) [reply awatif-ui/src/Viewer.ts]: Yeah this is much cleaner. I've pushed those changes.

**Pull request merged**